Support `system.vendor-name` hardware requirement for `mrack`

skycastlelily opened the pull request:
Pull Request Checklist

* [x] implement the feature
* [ ] write the documentation
* [x] extend the test coverage
* [x] update the specification
* [ ] adjust plugin docstring
* [ ] modify the json schema
* [ ] mention the version
* [x] include a release note

therazix commented on spec/hardware/system.fmf:
```diff
@@ -42,4 +45,4 @@ example:
 link:
   - implemented-by: /tmt/steps/provision/mrack.py
-    note: "``system.numa-nodes`` and ``system.model-name`` only"
+    note: "``system.vendor`` and ``system.model`` not implemented yet"
```
Perhaps I'm misunderstanding, but shouldn't it now be implemented along with `system.numa-nodes`?
```suggestion
    note: "``system.numa-nodes``, ``system.vendor`` and ``system.model`` only"
```

therazix commented on tmt/steps/provision/mrack.py:
```diff
@@ -596,6 +596,23 @@ def _transform_system_model_name(
 
 
+def _transform_system_vendor_name(
+        constraint: tmt.hardware.NumberConstraint,
```
Shouldn't it be `TextConstraint` instead of  `NumberConstraint`?
```suggestion
        constraint: tmt.hardware.TextConstraint,
```

skycastlelily replied:
yeah, sorry for failing to notice that by myself,and update:)

happz commented on spec/hardware/system.fmf:
```diff
@@ -20,6 +20,9 @@ description: |
            numa-nodes: 2|">= 2"
 
+    .. versionchanged:: 1.39
+       ``beaker`` plugins supports ``vendor-name``
```
`plugin`, not `plugins`

thrix replied:
resolved with an update

skycastlelily commented:
Rebased and updated, and it just occurred to me that I should update first
in order to put the code addressing the comments in one commit, then
rebase, will keep that in mind^^

On Sat, Nov 2, 2024 at 5:22 PM Miloš Prchlík ***@***.***>
wrote:

> ***@***.**** approved this pull request.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/teemtee/tmt/pull/3314#pullrequestreview-2411221507>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AKFR23GC3BOXXC7GCOLREPDZ6SKUXAVCNFSM6AAAAABQQUSMLSVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDIMJRGIZDCNJQG4>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

skycastlelily commented:
Rebased and updated^^

thrix commented:
/packit test
